### DISCUSSION TITLE
Pre-selecting specific rows?

### DISCUSSION THREAD

**Discussion by luckman212** [junegunn/fzf#2658] [Q&A]
@junegunn 

Is there any way to preselect 1 or more rows when `fzf` is invoked?

e.g. in the below, I was hoping to have fzf appear with `apple` and `peach` selected:
```shell
echo "apple|banana|peach|pear" | tr '|' '\n' |
fzf --multi
```

![image](https://user-images.githubusercontent.com/1992842/141307542-52d935a1-062f-44be-82be-a0bd37ad0e72.png)

eg. something like
```
echo "apple|banana|peach|pear" | tr '|' '\n' |
fzf --multi --preselect='apple|peach'
```

Would be amazing if it supported regex or glob style matches... e.g. the below would match `peach` and `pear`
```
echo "apple|banana|peach|pear" | tr '|' '\n' |
fzf --multi --preselect='^pe.*'
```
![image](https://user-images.githubusercontent.com/1992842/141307742-b668fac7-8ffb-4915-9ae1-f64a35044e7f.png)

**Comment by luckman212** [comment]
Safe to assume the answer is **_no_**. Please consider this as a FR !

**Comment by TobiaszCudnik** [comment]
This works for selecting the 2nd row on start:
> --bind "start:pos(2)"

You can probably chain `pos` and `select` actions to achieve a multi-pre-select.

**Comment by LangLangBart** [comment]
Based on the provided examples, it seems that the existing options and search syntax already support basic selections[^1].



> echo "apple|banana|peach|pear" | tr '|' '\n' |
> fzf --multi --preselect='apple|peach'


```bash
echo -e "apple\nbanana\npeach\npear" |
	fzf --multi --query 'apple | peach' --bind 'load:select-all+clear-query'
```

> echo "apple|banana|peach|pear" | tr '|' '\n' |
> fzf --multi --preselect='^pe.*'


```bash
echo -e "apple\nbanana\npeach\npear" |
	fzf --multi --query '^pe' --bind 'load:select-all+clear-query'
```

[^1]: [README.md - Search Syntax](https://github.com/junegunn/fzf?tab=readme-ov-file#search-syntax)

**Comment by Diegovsky** [comment]
This doesn't work if lines can contain similar characters of if the regex is greater than the max allowed characters, like here:
![image](https://github.com/junegunn/fzf/assets/46163903/7bbf3454-4e27-41a9-a96a-2324032d110e)

An actual solution would be preferable instead of this big hack.

**Reply by LangLangBart** [reply]
> This doesn't work if lines can contain similar characters

Can you share the command? Did you try the `--exact` flag or `'`?

```sh
# fuzzy match A*r*c*h
man -k . 2>/dev/null | fzf --query "Arch" --bind 'result:unbind(result)+select-all'
# 143 results

# extra ' to match Arch
man -k . 2>/dev/null | fzf --query "'Arch" --bind 'result:unbind(result)+select-all'
# 7 results
```
